source/extensions/filters/http/admission_control/evaluators/success_criteria_evaluator.cc

```diff
@@ -0,0 +1,74 @@
+#include "extensions/filters/http/admission_control/evaluators/success_criteria_evaluator.h"
+
+#include <algorithm>
+
+#include "envoy/common/exception.h"
+#include "envoy/grpc/status.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/common/fmt.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AdmissionControl {
+
+SuccessCriteriaEvaluator::SuccessCriteriaEvaluator(
+    envoy::extensions::filters::http::admission_control::v3alpha::AdmissionControl::SuccessCriteria
+        success_criteria) {
+  // HTTP status.
+  if (success_criteria.has_http_criteria()) {
+    for (const auto& range : success_criteria.http_criteria().http_success_status()) {
+      if (!validHttpRange(range.start(), range.end())) {
+        throw EnvoyException(
+            fmt::format("invalid HTTP range: [{}, {})", range.start(), range.end()));
+      }
+
+      http_success_fns_.emplace_back([range](uint64_t status) {
+        return (static_cast<uint64_t>(range.start()) <= status) &&
+               (status < static_cast<uint64_t>(range.end()));
+      });
```

Comment:
what do you think about doing the casts outside the lambda and capture the two `uint64_t` instead? might make this statement easier to read